### ISSUE TITLE
Fails to install without "git"

### ISSUE THREAD

**Issue by tianon** [hylang/hy#1205]
With the newly released 0.12.0 (congrats!), if `git` is installed, all works fine (as expected), but if `git` is not installed, the installation fails:
```console
$ docker run -it --rm python:3.6-slim bash
root@fb6e7113d667:/# pip install hy
Collecting hy
  Downloading hy-0.12.0.tar.gz (63kB)
    100% |████████████████████████████████| 71kB 1.6MB/s 
    Complete output from command python setup.py egg_info:
    Traceback (most recent call last):
      File "<string>", line 1, in <module>
      File "/tmp/pip-build-3xk7ny_l/hy/setup.py", line 35, in <module>
        (["git", "describe", "--tags", "--dirty"])
      File "/usr/local/lib/python3.6/subprocess.py", line 336, in check_output
        **kwargs).stdout
      File "/usr/local/lib/python3.6/subprocess.py", line 403, in run
        with Popen(*popenargs, **kwargs) as process:
      File "/usr/local/lib/python3.6/subprocess.py", line 707, in __init__
        restore_signals, start_new_session)
      File "/usr/local/lib/python3.6/subprocess.py", line 1326, in _execute_child
        raise child_exception_type(errno_num, err_msg)
    FileNotFoundError: [Errno 2] No such file or directory: 'git'
    
    ----------------------------------------
Command "python setup.py egg_info" failed with error code 1 in /tmp/pip-build-3xk7ny_l/hy/
```

This behavior appears to have been introduced in https://github.com/hylang/hy/pull/1135, which looks like it accounts for `git` failing, but not for `git` not being installed at all. :sweat_smile:

**Comment by Kodiologist** [comment]
How does [this patch](https://github.com/Kodiologist/hy/commit/a98fabecb2145b526ce9e0d441d6dbb90bd4aeb7) work for you?

**Comment by tianon** [comment]
Seems to work OK: (although @kirbyfan64 looks to have some suggestion for improvement)
```console
$ docker run -it --rm python:3.6-slim bash
root@b011412a2806:/# pip install https://github.com/Kodiologist/hy/archive/version-gen-updates.tar.gz
Collecting https://github.com/Kodiologist/hy/archive/version-gen-updates.tar.gz
  Downloading https://github.com/Kodiologist/hy/archive/version-gen-updates.tar.gz (376kB)
    100% |████████████████████████████████| 378kB 2.0MB/s 
Collecting rply>=0.7.0 (from hy===unknown)
  Downloading rply-0.7.4-py2.py3-none-any.whl
Collecting astor>=0.5 (from hy===unknown)
  Downloading astor-0.5-py2.py3-none-any.whl
Collecting clint>=0.4 (from hy===unknown)
  Downloading clint-0.5.1.tar.gz
Collecting appdirs (from rply>=0.7.0->hy===unknown)
  Downloading appdirs-1.4.0-py2.py3-none-any.whl
Collecting args (from clint>=0.4->hy===unknown)
  Downloading args-0.1.0.tar.gz
Installing collected packages: appdirs, rply, astor, args, clint, hy
  Running setup.py install for args ... done
  Running setup.py install for clint ... done
  Running setup.py install for hy ... done
Successfully installed appdirs-1.4.0 args-0.1.0 astor-0.5 clint-0.5.1 hy-unknown rply-0.7.4
```

**Comment by Kodiologist** [comment]
Fixed by #1209.
